crates/matrix-sdk-sqlite/src/crypto_store.rs

```diff
@@ -1232,6 +1275,31 @@ impl CryptoStore for SqliteCryptoStore {
     }
 }
 
+fn generate_placeholders(number: usize) -> String {
+    if number == 0 {
+        panic!("Can't generate zero placeholders");
+    }
+    (std::iter::repeat("?").take(number)).join(", ")
+}
```

Comment:
Just like for `chunk_large_query_over`, we have a similar API: [`repeat_vars`](https://github.com/matrix-org/matrix-rust-sdk/blob/18b6387a7ad8d24ca258a4ec92d6d4f7f20fa4e5/crates/matrix-sdk-sqlite/src/state_store.rs#L1654-L1659) that you can re-use :-).

Reply:
Cool, will refactor on top of `chunk_large_query_over`